packages/store/types/tests/state-operators.append.lint.ts

```diff
@@ -15,23 +15,22 @@ describe('[TEST]: the append State Operator', () => {
       bools: [true, false]
     };
 
-    patch<Original>({ nums: append<number>([]) })(original); // $ExpectType { nums: number[]; strs: string[]; bools: boolean[]; }
-    patch<Original>({ nums: append(null!) })(original); // $ExpectType { nums: number[]; strs: string[]; bools: boolean[]; }
-    patch<Original>({ nums: append(undefined!) })(original); // $ExpectType { nums: number[]; strs: string[]; bools: boolean[]; }
-    patch<Original>({ nums: append([1, 2]) })(original); // $ExpectType { nums: number[]; strs: string[]; bools: boolean[]; }
+    patch<Original>({ nums: append<number>([]) })(original); // $ExpectType Original
+    patch<Original>({ nums: append(null!) })(original); // $ExpectType Original
+    patch<Original>({ nums: append(undefined!) })(original); // $ExpectType Original
+    patch<Original>({ nums: append([1, 2]) })(original); // $ExpectType Original
```

Comment:
Changing the returned expected type here may be a breaking change on patch. We need explicit dtslint tests for the `patch` operator if we are going to make this change. I will add the tests for the `patch` and `compose` operators tonight.
I have been very strict with myself following the rule that we discussed a few months back. Any typings change needs to have thorough typings tests documenting all aspects of the existing behavior before any change is done to the typings in a subsequent PR. This is why I have been solving these typings issue step by step and not all in one go.

Reply:
I could not do it in a less complex way, however, the return type, the correct one, is quite logical.

patch<T> => return T

Reply:
This may interfere with the implicit operators. I will investigate if my suspicion is true. I hope I am wrong because what you have above does feel better. :+1: